controller/scheduler-v2/formation.go

```diff
@@ -0,0 +1,234 @@
+package main
+
+import (
+	"fmt"
+	"math"
+
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/controller/utils"
+	"github.com/flynn/flynn/host/types"
+)
+
+type Formations struct {
+	formations map[utils.FormationKey]*Formation
+}
```

Comment:
This could just be `type Formations map[utils.FormationKey]*Formation` (then use `make(Formations)` rather than `newFormations()`)